src/nmodl/parse1.y

```diff
@@ -1221,8 +1222,14 @@ nrnuse: USEION NAME READ nrnlist valence
 		{nrn_use($2, ITEM0, $4, $5);}
 	|USEION NAME READ nrnlist WRITE nrnlist valence
 		{nrn_use($2, $4, $6, $7);}
+	|USEION NAME READ nrnlist valence REPRESENTS ONTOLOGY_ID
+		{nrn_use($2, $4, ITEM0, $5);}
+	|USEION NAME WRITE nrnlist valence REPRESENTS ONTOLOGY_ID
+		{nrn_use($2, ITEM0, $4, $5);}
+	|USEION NAME READ nrnlist WRITE nrnlist valence REPRESENTS ONTOLOGY_ID
+		{nrn_use($2, $4, $6, $7);}
 	|USEION error
-		{myerr("syntax is: USEION ion READ list WRITE list");}
+		{myerr("syntax is: USEION ion READ list WRITE list REPRESENTS cui");}
 	;
```

Comment:
could be simplified bit by adding one new production (my local changes to this PR as below) : 

```
+%type  <qp>    neuronblk nrnuse nrnlist optnrnlist valence initstmt bablk optontology
 %token <qp>    CONDUCTANCE
 %type  <qp>    conducthint

@@ -1216,17 +1216,11 @@ nrnstmt: /*nothing*/
        | nrnstmt THREADSAFE optnrnlist
                { threadsafe_seen($2, $3); }
        ;
-nrnuse: USEION NAME READ nrnlist valence
+nrnuse: USEION NAME READ nrnlist valence optontology
                {nrn_use($2, $4, ITEM0, $5);}
-       |USEION NAME WRITE nrnlist valence
+       |USEION NAME WRITE nrnlist valence optontology
                {nrn_use($2, ITEM0, $4, $5);}
-       |USEION NAME READ nrnlist WRITE nrnlist valence
-               {nrn_use($2, $4, $6, $7);}
-       |USEION NAME READ nrnlist valence REPRESENTS ONTOLOGY_ID
-               {nrn_use($2, $4, ITEM0, $5);}
-       |USEION NAME WRITE nrnlist valence REPRESENTS ONTOLOGY_ID
-               {nrn_use($2, ITEM0, $4, $5);}
-       |USEION NAME READ nrnlist WRITE nrnlist valence REPRESENTS ONTOLOGY_ID
+       |USEION NAME READ nrnlist WRITE nrnlist valence optontology
                {nrn_use($2, $4, $6, $7);}
        |USEION error
                {myerr("syntax is: USEION ion READ list WRITE list REPRESENTS cui");}
@@ -1252,6 +1246,9 @@ valence: /*nothing*/
                delete($2); lastok = $3;
                }
        ;
+optontology: { $$ = NULL; }
+           | REPRESENTS ONTOLOGY_ID
+             { $$ = $2; }
 %%
```

Reply:
I think I meant CURIE instead of CUI in the error?